File: 4014-minimum-total-price-after-applying-discounts/4014-minimum-total-price-after-applying-discounts.java
class Solution {
    public double minPrice(int[] prices, int[] discounts) {
        Arrays.sort(prices);
        Arrays.sort(discounts);
        int n = prices.length, m = discounts.length;
        double ans = 0;
        int j = m - 1;
        for (int i = n - 1; i >= 0; i--) {
            if (j >= 0) {
                double price = prices[i] * (1 - 0.01 * discounts[j]);
                ans += price;
                j--;
            } else
                ans += prices[i];
        }
        return ans;
    }
}

File: 4014-minimum-total-price-after-applying-discounts/4014-minimum-total-price-after-applying-discounts.cpp
class Solution {
public:
    double minPrice(vector<int>& prices, vector<int>& discounts) {
        double ans = 0;
        int n = prices.size(), m = discounts.size();
        sort(prices.begin(), prices.end(), greater<int>());
        sort(discounts.begin(), discounts.end(), greater<int>());
        int j = 0;
        for (int i = 0; i < n; i++) {
            if (j < m) {
                double price = prices[i] * (1 - 0.01 * discounts[j]);
                ans += price;
                j++;
            } else
                ans += (double)prices[i];
        }
        return ans;
    }
};